rfcs/experimental/dynamic_selection/README.md

```diff
@@ -0,0 +1,516 @@
+```
+
+| Constructors and Initialization |
+| -----------------------------------------------------------|
+| `auto_tune_policy(deferred_initialization_t); // (1)` |
+| `auto_tune_policy(double resample_time=never_resample); // (2)` |
+| `auto_tune_policy(const std::vector<resource_t<Backend>>& resources, double resample_time=never_resample); // (3)` |
+| `void initialize(double resample_time=never_resample); // (4)` |
+| `void initialize(const std::vector<resource_t<Backend>>& resources, double resample_time=never_resample); // (5)` |
+
+1. Defers initialization and requires a later call to `initialize`.
+2. Profiles each set of unique `f` and `args` for each resource in the default set and then uses the best.
+3. Uses the provided set of resources and profiles each set of unique `f` and `args` for each resource uses the best.
+4. Profiles each set of unique `f` and `args` for each resource in the default set and then uses the best.
+5. Uses the provided set of resources and profiles each set of unique `f` and `args` for each resource uses the best.
+
+`initialize` should only be called for policies that are constructed with `deferred_initialization_t`.
+
+#### select heuristic (expository)
+
+```cpp
+  template<typename Function, typename ...Args>
+  selection_type select(Function&& f, Args&&...args) {
+    if (initialized_) {
+      auto k = make_task_key(f, args...);
+      auto tuner = get_tuner(k);
+      auto offset = tuner->get_resource_to_profile();
+      if (offset == use_best) {
+        return selection_type {*this, tuner->best_resource_, tuner}; 
+      } else {
+        auto r = resources_[offset];
+        return selection{*this, r, tuner}; 
+      }
+    } else {
+      throw std::logic_error(“select called before initialization”);
+    } 
+  }
+```
+
+#### execution info reporting requirements
+
+```cpp
+  void report(task_execution_time_t, timing_t v) { 
+    tuner_->add_timing(offset_, v); 
+  }
+```
+
+#### Exceptions
+
+Constructor or initialize may throw `std::bad_alloc` or `std::logic_error`.
+
+<a id="backend_req_id"></a>
+## Backends
+
+Backends allow generic policies to be implemented. End-users do not directly interact
+with a backend, except to choose the backend if they opt-out of the default SYCL backend.
+Custom policy writers that wish to implement a generic policy that can accept backends
+should follow the backend contract. Developers that wish to provide a backend that can
+be used with the existing concrete policies should also follow the backend contract.
+
+### Named Requirements
+
+The type `T` satisfies the *Backend* contract if given,
+
+- `b` an arbitrary identifier of type `T`
+- `args` an arbitrary parameter pack of types `typename… Args`
+- `s` is of type `S` and satisfies *Selection* and `is_same_v<resource_t<S>, resource_t<T>>` is `true`
+- `f` a function object with signature `wait_t<T> fun(resource_t<T>, Args…);`
+- `r` a container of resources of type `std::vector<resource_t<T>`.
```

Comment:
`r` is not used anywhere in the description of the backend contract.

Reply:
Will remove.